app/src/main/java/org/p2p/wallet/home/ui/new/NewSelectTokenFragment.kt

```diff
@@ -0,0 +1,113 @@
+package org.p2p.wallet.home.ui.new
+
+import androidx.appcompat.widget.SearchView
+import androidx.appcompat.widget.Toolbar
+import androidx.core.os.bundleOf
+import androidx.core.view.isVisible
+import androidx.fragment.app.setFragmentResult
+import androidx.recyclerview.widget.LinearLayoutManager
+import android.os.Bundle
+import android.view.View
+import org.koin.android.ext.android.inject
+import org.p2p.uikit.utils.attachAdapter
+import org.p2p.uikit.utils.showSoftKeyboard
+import org.p2p.wallet.R
+import org.p2p.wallet.common.analytics.interactor.ScreensAnalyticsInteractor
+import org.p2p.wallet.common.mvp.BaseMvpFragment
+import org.p2p.wallet.databinding.FragmentSelectTokenNewBinding
+import org.p2p.wallet.home.model.SelectTokenItem
+import org.p2p.wallet.home.model.Token
+import org.p2p.wallet.home.ui.new.adapter.NewSelectTokenAdapter
+import org.p2p.wallet.moonpay.analytics.BuyAnalytics
+import org.p2p.wallet.utils.args
+import org.p2p.wallet.utils.popBackStack
+import org.p2p.wallet.utils.unsafeLazy
+import org.p2p.wallet.utils.viewbinding.viewBinding
+import org.p2p.wallet.utils.withArgs
+
+private const val ARG_ALL_TOKENS = "ARG_ALL_TOKENS"
+private const val ARG_SELECTED_TOKEN = "ARG_SELECTED_TOKEN"
+private const val ARG_REQUEST_KEY = "ARG_REQUEST_KEY"
+private const val ARG_RESULT_KEY = "ARG_RESULT_KEY"
+
+class NewSelectTokenFragment :
+    BaseMvpFragment<NewSelectTokenContract.View, NewSelectTokenContract.Presenter>(R.layout.fragment_select_token_new),
+    NewSelectTokenContract.View,
+    SearchView.OnQueryTextListener {
+
+    companion object {
+        fun create(tokens: List<Token.Active>, selectedToken: Token.Active?, requestKey: String, resultKey: String) =
```

Comment:
selectedToken  -> preSelectedToken?

Reply:
It doesn't make any difference. 
current naming is full of context and the developer will understand what is this token about. 
Why we need to make the name longer?

Reply:
maybe makes sense, yes.
just preSelected gives more context that it was selected before we open SelectToken and we render fragment with such context

Reply:
imho that's overhead. 
1. We need to make names shorter to write less
2. We need to name it to allow the other dev to understand what is going on. `selectedToken` looks absolute understandable. 
3. Imho, there is no need to add more context and make the name longer. 
4. That's not a problem, that's not an error, that's not a context absence.  

@kabramovp2p @davran1001 do you think there is additional context needed in naming?